assets/blocks/course-outline/edit.js

```diff
@@ -85,8 +98,39 @@ const EditCourseOutlineBlock = ( {
 	);
 };
 
-export default withSelect( ( select ) => {
-	return {
-		structure: select( COURSE_STORE ).getStructure(),
+const applyWithSelect = withSelect( ( select ) => ( {
+	structure: select( COURSE_STORE ).getStructure(),
+} ) );
+
+const applyWithDispatch = withDispatch( ( dispatch, ownProps, { select } ) => {
+	const editPost = select( 'core/edit-post' );
+	const { toggleEditorPanelEnabled } = dispatch( 'core/edit-post' );
+
+	const hasLegacyMetaBoxes =
+		editPost.isEditorPanelEnabled( 'meta-box-course-lessons' ) ||
+		editPost.isEditorPanelEnabled( 'meta-box-module_course_mb' );
+
+	const toggleLegacyMetaboxes = ( enable ) => {
+		// Side-effect - Prevent submit modules.
+		document
+			.querySelectorAll( '#module_course_mb input' )
+			.forEach( ( input ) => {
+				input.disabled = ! enable;
+			} );
+
+		// If the `enable` is already as requested, do nothing.
+		if ( enable === hasLegacyMetaBoxes ) {
+			return;
+		}
+
+		toggleEditorPanelEnabled( 'meta-box-course-lessons' );
+		toggleEditorPanelEnabled( 'meta-box-module_course_mb' );
 	};
-} )( EditCourseOutlineBlock );
+
+	return { toggleLegacyMetaboxes };
+} );
+
+export default compose(
+	applyWithSelect,
+	applyWithDispatch
+)( EditCourseOutlineBlock );
```

Comment:
This might look better like this, and maybe a bit more clear:
```
compose(
	withSelect(selectors),
	withDispatch(dispatches)
)( EditCourseOutlineBlock );
```
(with those two being just the inner functions, like `const dispatches = ( dispatch, ownProps, { select } ) => { ... }`)

Reply:
Thank you for the review! Updated here: https://github.com/Automattic/sensei/pull/3649/commits/2bacb33a193e7350a1f686bf79a026931b51c454